lib/generate-index.js

```diff
@@ -38,7 +38,19 @@ function generateIndex (playbook, pages) {
     // Exclude pages marked as "noindex"
     .filter(({ page, $ }) => {
       const $metaRobots = $('meta[name=robots]')
-      return !(($metaRobots && $metaRobots.attr('content') === 'noindex') || (page.asciidoc && page.asciidoc.attributes && page.asciidoc.attributes.noindex === ''))
+      const indexable = !(($metaRobots && $metaRobots.attr('content') === 'noindex') ||
+                        (page.asciidoc && page.asciidoc.attributes && page.asciidoc.attributes.noindex === ''))
+      const indexOnlyLatest = env.DOCSEARCH_INDEX_VERSION &&
+                              env.DOCSEARCH_INDEX_VERSION === 'latest'
+
+      if (indexOnlyLatest) {
+        const component = contentCatalog.getComponent(page.src.component)
+        const thisVersion = contentCatalog.getComponentVersion(component, page.src.version)
+        const latestVersion = contentCatalog.getComponent(page.src.component).latest
+        return indexable && thisVersion === latestVersion
+      } else {
+        return indexable
+      }
```

Comment:
Since you are using `return` in the `if` branch, the `else` branch is superfluous:

```suggestion
      }
      return indexable
```